perceval/converters/abstract_converter.py

```diff
@@ -0,0 +1,147 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from abc import ABC, abstractmethod
+
+from perceval.components import Port, Circuit, Processor, Source
+from perceval.utils import P, BasicState, Encoding
+from perceval.utils.algorithms.optimize import optimize
+from perceval.utils.algorithms.norm import frobenius
+import perceval.components.unitary_components as comp
+
+MIN_PRECISION_GATE = 1e-4
+
+
+def _create_mode_map(c_idx: int, c_data: int) -> dict:
+    return {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+
+
+class AGateConverter(ABC):
+    r"""
+    Converter class for gate based Circuits to perceval processor
+    Qiskit or MyQLM
+    """
+
+    def __init__(self, catalog, **kwargs):
```

Comment:
Since `MyQLMConverter` and `QiskitConverter` `__init__` methods share exactly the same signature (i.e. same parameters, same order, same names), I don't know why you went for **kwargs.
**kwargs is a solution for signature variability, but here, signatures are exactly the same.